sdk/identity/identity/src/client/identityClient.ts

```diff
@@ -301,10 +309,51 @@ export class IdentityClient extends ServiceClient implements INetworkModule {
     });
 
     const response = await this.sendRequest(request);
+
+    this.logIdentifiers(response);
+
     return {
       body: response.bodyAsText ? JSON.parse(response.bodyAsText) : undefined,
       headers: response.headers.toJSON(),
       status: response.status,
     };
   }
+
+  /**
+   * If allowLoggingAccountIdentifiers was set on the constructor options
+   * we try to log the account identifiers by parsing the received access token.
+   *
+   * The account identifiers we try to log are:
+   * - `appid`: The application or Client Identifier.
+   * - `upn`: User Principal Name.
+   *   - It might not be available in some authentication scenarios.
+   *   - If it's not available, we put a placeholder: "No User Principal Name available".
+   * - `tid`: Tenant Identifier.
+   * - `oid`: Object Identifier of the authenticated user.
+   */
+  private logIdentifiers(response: PipelineResponse): void {
+    if (!this.allowLoggingAccountIdentifiers || !response.bodyAsText) {
+      return;
+    }
+    const unavailableUpn = "No User Principal Name available";
+    try {
+      const parsed = (response as any).parsedBody || JSON.parse(response.bodyAsText);
```

Comment:
Would it be possible to get any of this info from MSAL in a way that doesn't require us to parse the raw response? I'm a bit concerned that we could somehow unintentionally log the wrong sensitive data.

Reply:
As long as we limit ourselves to these 4 values we should be ok to not log something too sensitive.

MSAL does allow you to retrieve this with their AccountRecord, but we don’t use MSAL in every credential.